api/controllers/UserController.js

```diff
@@ -914,81 +892,77 @@ module.exports = {
-    if (record.emailIndex(email) !== -1) {
+    if (user.emailIndex(emailToAdd) !== -1) {
       logger.warn(
-        `[UserController->addEmail] Email ${email} already belongs to ${userId}.`,
+        `[UserController->addEmail] Email ${emailToAdd} already belongs to ${userId}.`,
         {
           request,
           fail: true,
           user: {
             id: userId,
-            email: record.email,
+            email: user.email,
           },
         },
       );
       throw Boom.badRequest('Email already exists');
     }
 
     // Make sure email added is unique to the entire HID system.
-    const erecord = await User.findOne({ 'emails.email': email });
+    const erecord = await User.findOne({ 'emails.email': emailToAdd });
     if (erecord) {
       logger.warn(
-        `[UserController->addEmail] Email ${email} is not unique`,
+        `[UserController->addEmail] Email ${emailToAdd} is not unique`,
         {
           request,
           fail: true,
           user: {
             id: userId,
-            email: record.email,
+            email: user.email,
           },
         },
       );
       throw Boom.badRequest('Email is not unique');
     }
 
     const data = {
-      email,
+      email: emailToAdd,
       type: 'Work',
       validated: false,
     };
-    record.emails.push(data);
-    record.lastModified = new Date();
+    user.emails.push(data);
+    user.lastModified = new Date();
 
-    const savedRecord = await record.save();
+    await user.save();
     logger.info(
-      `[UserController->addEmail] Successfully saved user ${record.id}`,
+      `[UserController->addEmail] Successfully saved user ${user.id}`,
       {
         request,
         user: {
-          id: record.id,
+          id: user.id,
         },
       },
     );
-    const savedEmailIndex = savedRecord.emailIndex(email);
-    const savedEmail = savedRecord.emails[savedEmailIndex];
+    const savedEmailIndex = user.emailIndex(emailToAdd);
+    const savedEmailId = user.emails[savedEmailIndex]._id.toString();
 
     // Send confirmation email
-    await EmailService.sendValidationEmail(
-      record,
-      email,
-      savedEmail._id.toString(),
-      appValidationUrl,
-    );
+    await EmailService.sendValidationEmail(user, emailToAdd, savedEmailId);
 
-    // If the email sent without error, notify the other emails on this account.
+    // If the email sent without error, notify the other confirmed emails on
+    // this account.
     const promises = [];
-    for (let i = 0; i < record.emails.length; i++) {
-      // TODO: probably shouldn't send notices to unconfirmed email addresses.
-      //
-      // @see HID-2150
-      promises.push(EmailService.sendEmailAlert(record, record.emails[i].email, email));
+    for (let i = 0; i < user.emails.length; i++) {
+      const thisEmail = user.emails[i];
+      if (thisEmail.validated) {
+        promises.push(EmailService.sendEmailAlert(user, thisEmail.email, emailToAdd));
+      }
```

Comment:
This is the core fix to the ticket. There were lots of other standardizations I made (e.g. changing `record` to `user` and renaming the very easy-to-confuse `email` variable to be more descriptive).